In [1]:
import requests
import json
import os

### 1. The Consumer Expenditure (CEX) survey is conducted by the Bureau of Labor Statistics (BLS). From the web-page1 of the CEX survey, answer the following questions:


#### 1. What is the definition of the target population for this survey? (These are called: primary sampling units (PSUs))
- Selection of nationwide households representing the civilian noninstitutional population
- includes people living in houses, condominiums, apartments, and group quarters such as college dormitories

#### 2. What is the purpose of the CEX survey?
- effort to obtain information on the complete range of consumers’ expenditures, income, and demographic characteristics, in the same survey, directly from consumers
- to provide weights for the Consumer Price Index (CPI)

#### 3. If taken in its entirety, what type of panel data will you have?

- **Repeated cross section**: several sets of cross sectional data taken at consecutive time periods.

### 2. The API key that you obtained (a string of 32 letters and numbers) can be used to retrieve data from their API in the following way:
The base URL is https://api.stlouisfed.org/ The endpoint you
will use is: fred/series/observations The parameters you will pass
(after a question mark symbol and separated with & symbol) are: ?series_id=GDP&api_key=8cbc8c272d6ecc3c548465468f5e90c4&observation_ start=2021-01-01&file_type=json. For example, if your API key is abcdefghijklmnopqrstuvwxyz123456, you will make the following call: https://api.stlouisfed.org/fred/series/observations?series_id=GDP&api_key=abcdefghijklmnopqrstuvwxyz123456&observation_start=2021-01-01&file_type=json Your code should show how the
URL is created (suppress the key). Your notebook should show the response you get.

#### 1. Obtain an API key for the FRED data set. Go to https://fred. stlouisfed.org/docs/api/api_key.html and click on the Request or view your API keys link. You will create an acount and request an API key.

In [2]:
%run APIkeys.py

API keys loaded


#### 2. The API key that you obtained (a string of 32 letters and numbers) can be used to retrieve data from their API in the following way:

In [3]:
baseURL = 'https://api.stlouisfed.org/'
endPoint = 'fred/series/observations'
parameter = 'series_id=GDP'
parameter2 = 'observation_start=2021-01-01&file_type=json'

key = os.environ['FRED_API_key']

url = baseURL + endPoint + '?' + parameter + '&' + 'api_key=' + key + '&' + parameter2

In [4]:
print(url)

https://api.stlouisfed.org/fred/series/observations?series_id=GDP&api_key=71d28cbb276869ddcf3fa5c6fadf0a3b&observation_start=2021-01-01&file_type=json


In [5]:
response = requests.get(url)

In [6]:
print(response)

<Response [200]>


In [7]:
data = response.json()

In [8]:
data

{'realtime_start': '2022-09-21',
 'realtime_end': '2022-09-21',
 'observation_start': '2021-01-01',
 'observation_end': '9999-12-31',
 'units': 'lin',
 'output_type': 1,
 'file_type': 'json',
 'order_by': 'observation_date',
 'sort_order': 'asc',
 'count': 6,
 'offset': 0,
 'limit': 100000,
 'observations': [{'realtime_start': '2022-09-21',
   'realtime_end': '2022-09-21',
   'date': '2021-01-01',
   'value': '22038.226'},
  {'realtime_start': '2022-09-21',
   'realtime_end': '2022-09-21',
   'date': '2021-04-01',
   'value': '22740.959'},
  {'realtime_start': '2022-09-21',
   'realtime_end': '2022-09-21',
   'date': '2021-07-01',
   'value': '23202.344'},
  {'realtime_start': '2022-09-21',
   'realtime_end': '2022-09-21',
   'date': '2021-10-01',
   'value': '24002.815'},
  {'realtime_start': '2022-09-21',
   'realtime_end': '2022-09-21',
   'date': '2022-01-01',
   'value': '24386.734'},
  {'realtime_start': '2022-09-21',
   'realtime_end': '2022-09-21',
   'date': '2022-04-01',
   '

#### 3. Looking at your answer above, what is the US GDP in the first quarter of 2022? (units?)


$ 24386.73 billion